### ISSUE TITLE
DicomFilter creates a buffer that DicomFile/Element cannot parse

### ISSUE THREAD

**Issue by cppcooper** [BCCF-UBCO-AD/Orthanc-TMI#82]
It is unclear, currently, why the filtered buffer appears malformed when we parse it for verification.

What is known, is the problem seems to be at the end of the buffer. Here are the last 3 parsed elements of before and after filtering.
Before:
```
[4F] (fffe,e000)->(e000fffe)
 idx: 1666426, next: 1684406, size: 17980, value offset: 8, length: 17972
[8F] (fffe,e000)->(e000fffe)
 idx: 1684406, next: 1702390, size: 17984, value offset: 8, length: 17976
[] (fffe,e0dd)->(e0ddfffe)
 idx: 1702390, next: 1702398, size: 8, value offset: 8, length: 0
 ```
 After:
 ```
 [ F] (fffe,e000)->(e000fffe)
 idx: 1648420, next: 1666380, size: 17960, value offset: 8, length: 17952
[4F] (fffe,e000)->(e000fffe)
 idx: 1666380, next: 1684360, size: 17980, value offset: 8, length: 17972
[8F] (fffe,e000)->(e000fffe)
 idx: 1684360, next: 1702344, size: 17984, value offset: 8, length: 17976
 ```
 As we can see, after it doesn't find the true last element.. it finds the second last element. So this is largely good/fine and might just be an off by one error like making the buffer 1 element too small so we get an error when we try to accessing the last byte.
 
edit: And here is the buffer copy procedure:
```
i: 0, range.1: 0, range.2: 2176, copy_size: 2176
i: 2176, range.1: 2212, range.2: 2228, copy_size: 16
i: 2192, range.1: 2238, range.2: 1702398, copy_size: 1700160
```

**Comment by cppcooper** [comment]
This issue affects #79

**Comment by cppcooper** [comment]
The resulting size calculated for the new buffer was incorrect, I did not take a look at examples but inferred the calculated size was too small for what was required. We can infer this based on the resulting corruption after the final (largest) copy into the destination buffer, if it were the same size or larger than what the copy needed there wouldn't have been an issue.

The calculation for this length was fairly convoluted, and no longer made sense looking at it again after a few weeks, so it was replaced. This has all been added onto PR #79 which is passing and can be merged now.
